api-reference/10 UI Components/dxCheckBox/1 Configuration/value.md

```diff
@@ -11,13 +11,13 @@ Specifies the UI component state.
 The check box can be in one of the following states depending on the **value** property's value.
 
 - checked  
- **value** is **true**
+ **value** is `true`
 
 - unchecked  
- **value** is **false**
+ **value** is `false`
 
-- undetermined  
- **value** is **undefined** or **null**
+- indeterminate  
+ **value** is `undefined` or `null`
 
 Users can press the **Space** key to change the **value**.
```

Comment:
This note makes me wonder if the following option name is good `allowIndeterminateStateByClick`. Maybe a better name would be `allowIndeterminateState`. Even "click" is not always a click - with all the mobile devices, it may be a tap.

Reply:
Me and developers had a long discussion about the name and we decided to use this one because when you simply say "allowIndeterminateState" - it's not true. It's already allowed. Like you already can set **value** to `null` or `undefined` and the CheckBox will be in the indeterminate state. It may confuse customers. They may think that we don't allow indeterminate state at all, and you need to enable the option to use the state, which is not true.
I think that clicking on a check box is the main way users interact with the component. Like yes, there are mobile devices, but when we use event handlers - we use onClick handler, not onClick and onTap. I also agree that yes, we can use the Space key to change the value, but it's also an additional way to change it, the main is still clicking.

Reply:
Also maybe as alternative - allowSwitchToIndeterminateState?